src/kakarot/instructions/system_operations.cairo

```diff
@@ -367,45 +324,30 @@ namespace SystemOperations {
     }(ctx: model.ExecutionContext*) -> model.ExecutionContext* {
         alloc_locals;
 
-        if (ctx.read_only != FALSE) {
+        if (ctx.call_context.read_only != FALSE) {
             let (revert_reason_len, revert_reason) = Errors.stateModificationError();
-            let ctx = ExecutionContext.revert(ctx, revert_reason, revert_reason_len);
+            let ctx = ExecutionContext.stop(ctx, revert_reason_len, revert_reason, TRUE);
             return ctx;
         }
 
-        // Get stack and memory from context
-        let stack = ctx.stack;
-
         // Stack input:
         // 0 - address: account to send the current balance to
-        let (stack, address_uint256) = Stack.pop(stack);
+        let (stack, address) = Stack.pop(ctx.stack);
+        let (_, address_high) = unsigned_div_rem(address.high, 2 ** 32);
```

Comment:
should we ensure the quotient is 0 or it's selfdestruct 's responsibility?

Reply:
according to @greged93 the address is just `address(uint160(popped))`, so no check

Reply:
should be ok this way, this is what [python specs do](https://github.com/ethereum/execution-specs/blob/master/src/ethereum/shanghai/utils/address.py#L25).